gpflux/layers/basis_functions/fourier_features/random/base.py

```diff
@@ -97,14 +110,32 @@ def _weights_build(self, input_dim: int, n_components: int) -> None:
             initializer=self._weights_init,
         )
 
-    def _weights_init(self, shape: TensorType, dtype: Optional[DType] = None) -> TensorType:
-        if isinstance(self.kernel, gpflow.kernels.SquaredExponential):
+    def _weights_init_individual(
+        self, kernel: gpflow.kernels.Kernel, shape: TensorType, dtype: Optional[DType] = None
+    ) -> TensorType:
+        if isinstance(kernel, gpflow.kernels.SquaredExponential):
             return tf.random.normal(shape, dtype=dtype)
         else:
-            p = _matern_number(self.kernel)
+            p = _matern_number(kernel)
             nu = 2.0 * p + 1.0  # degrees of freedom
             return _sample_students_t(nu, shape, dtype)
 
+    def _weights_init(self, shape: TensorType, dtype: Optional[DType] = None) -> TensorType:
+        if self.num_latent_gps is not None:
+            if isinstance(self.kernel, gpflow.kernels.SharedIndependent):
+                weights_list = [
+                    self._weights_init_individual(self.kernel.latent_kernels[0], shape[1:], dtype)
```

Comment:
Would it be possible for `_weights_init_individual` to broadcast over a batch dimension?

Reply:
I don't think so, for separateindependent you need to have some form of looping over the kernels themselves. Can you think of anything?

Reply:
Not the exact same - the samples end up being different